astropy/table/table.py

```diff
@@ -1638,6 +1716,19 @@ def add_columns(self, cols, indexes=None, copy=True, rename_duplicate=False):
                 new_indexes.insert(i, None)
                 newcols.insert(i, col)
 
+        if names is None:
+            names = ['col{}'.format(name) for name in range(len(self.columns), len(self.columns) + len(cols))]
+            for col, name in zip(cols, names):
+                if col.info.name is None:
+                    col.info.name = name
```

Comment:
I initially thought if col.info.name is None, i.e. if the column does not have a name, it can't really be a part of another table. Can a Table have unnamed columns?